base/inference.jl

```diff
@@ -542,7 +542,38 @@ add_tfunc(===, 2, 2,
         end
         return Bool
     end)
-add_tfunc(isdefined, 1, IInf, (args...)->Bool)
+function isdefined_tfunc(args...)
+    arg1 = args[1]
+    if isa(arg1, Const)
+        a1 = typeof(arg1.val)
+    else
+        a1 = widenconst(arg1)
+    end
+    if isType(a1)
+        return Bool
+    end
+    if isleaftype(a1)
+        if a1 <: Array # TODO
+        elseif a1 === Module # TODO
```

Comment:
I have the code for this case on https://github.com/JuliaLang/julia/pull/22281/files#diff-c0d983631c937ee5acfd0137cd7332deR2082. It's simple enough it's probably worth just including in this PR:
```julia
sym = args[2]
Symbol <: widenconst(sym) || return Bottom
if isa(sym, Const) && isa(sym.val, Symbol) && isdefined(a1, sym.val)
    return Const(true)
end
```